smart-contracts/contracts/cl-vault/src/vault/range/move_position.rs

```diff
@@ -321,11 +316,16 @@ pub fn do_swap_deposit_merge(
     } else {
         // if we have not tokens to swap, that means all tokens we correctly used in the create position
         // this means we can save the position id of the first create_position
-        POSITION.save(
+        // if position_id is None, then no first position was ever created, meaning we should never hit this else case.
+        let position_id = position_id.unwrap();
+        let modify = MODIFY_RANGE_STATE.load(deps.storage)?.unwrap();
```

Comment:
Idea - 
As we are now dealing with multi-range, Can we change MODIFY_RANGE_STATE internal struct ModifyRangeState to have current position id field and MODIFY_RANGE_STATE as vector of ModifyRangeState. That will help if we want to modify range for multiple positions at once.

Reply:
I think it's less complex to try to keep all actions only running on one position at a time, if we need the capability to move multiple at once, we can run the multiple execute msges in the same transaction